django/db/backends/base/schema.py

```diff
@@ -547,9 +547,7 @@ def _alter_field(self, model, old_field, new_field, old_type, new_type,
                 for fk_name in rel_fk_names:
                     self.execute(self._delete_constraint_sql(self.sql_delete_fk, new_rel.related_model, fk_name))
         # Removed an index? (no strict check, as multiple indexes are possible)
-        if (old_field.db_index and not new_field.db_index and
-                not old_field.unique and not
-                (not new_field.unique and old_field.unique)):
+        if old_field.db_index and not old_field.unique and (not new_field.db_index or new_field.unique):
```

Comment:
I find it fairly hard to grasp this change (and tbh, I have a hard time figuring out what was and what is going on). Care to extend the comment right above?